docs/data/date-pickers/digital-clock/digital-clock.md

```diff
@@ -0,0 +1,61 @@
+---
+product: date-pickers
+title: React Digital Clock component
+components: DigitalClock, MultiSectionDigitalClock
+githubLabel: 'component: TimePicker'
+packageName: '@mui/x-date-pickers'
+---
+
+# Digital Clock
+
+<p class="description">The Digital Clock components let the user select a time without any input or popper / modal.</p>
+
+## Description
+
+There are two component versions for different cases. The `DigitalClock` handles selection of a single time instance in one step, just like a `select` component. The `MultiSectionDigitalClock` allows selecting time using separate sections for separate views.
+
+The `DigitalClock` is more appropriate when there is a limited amount of time options needed, while the `MultiSectionDigitalClock` is suited for cases when a more granular time selection is needed.
+
+## Basic usage
+
+{{"demo": "DigitalClockBasic.js"}}
+
+## Uncontrolled vs. Controlled
+
+The components can be uncontrolled or controlled.
+
+{{"demo": "DigitalClockValue.js"}}
+
+## Form props
+
+The components can be disabled or read-only.
+
+{{"demo": "DigitalClockFormProps.js"}}
+
+## Views
+
+The `MultiSectionDigitalClock` component can contain three views: `hours`, `minutes`, and `seconds`.
```

Comment:
Yes, I'm aware of it. IMHO, once we implement it for fields, it would be just a matter of some small changes on the pickers' side 🤔 
Isn't this issue https://github.com/mui/mui-x/issues/5542 already intended for it? 🤔